### DISCUSSION TITLE
Can I create a short link with an expiration time?

### DISCUSSION THREAD

**Discussion by squidward05** [shlinkio/shlink#1964] [Help wanted] [answered]
### Shlink version

2.6.2

### PHP version

PHP 8.0.2

### How do you serve Shlink

RoadRunner Docker image

### Database engine

MySQL

### Database version

5.7.26

### Summary

In my scenario, a large number of short links are temporarily generated for clients, which leads to increasing bloat.Can I create a short link with an expiration time?

**Comment by acelaya** [comment]
Can you elaborate what "expiration time" means for you?

**Reply by squidward05** [reply]
For me, I need to generate a short link that is valid for one day. It can be automatically deleted when it expires, so that it does not occupy database or redis space, and there is no need to maintain a large number of existing short links.

**Reply by acelaya** [reply, marked as answer]
EDIT: As of Shlink 4.1.0, this will be more or less supported. Not fully automated, but Shlink will provide a CLI command to delete expired short URLs. See https://github.com/shlinkio/shlink/pull/2086 for details.

---

No, this is not supported at the moment.

It was [asked before](https://github.com/shlinkio/shlink/discussions/1710), but it seems I neglected to answer there for some reason.

**Reply by squidward05** [reply]
I think this is a very useful feature, it would be great if it could be implemented

**Reply by acelaya** [reply]
A variation of this might come eventually. See https://github.com/shlinkio/shlink/issues/2059

**Comment by ne0ark** [comment]
Is it possible to create links with default expiration times? Currently shlink chrome plugin doesn't allow you set it.
